src/os/bluestore/BlueStore.cc

```diff
@@ -7752,20 +7755,19 @@ void BlueStore::_txc_committed_kv(TransContext *txc)
     txc->onreadable_sync->complete(0);
     txc->onreadable_sync = NULL;
   }
-  unsigned n = txc->osr->parent->shard_hint.hash_to_shard(m_finisher_num);
   if (txc->oncommit) {
     logger->tinc(l_bluestore_commit_lat, ceph_clock_now() - txc->start);
-    finishers[n]->queue(txc->oncommit);
+    txc->oncommit->complete(0);
```

Comment:
that's what caused deadlocks when Sage last tested it with Teuthology. We need to leave that completions queued to finishers for now.